### PR TITLE
MES-4173 - Refactoring cat b fault summary code

### PR DESCRIPTION
**Pull request by TomBillingtonUK**
## Description
- Created a new function which can deal with counting Driving Faults, Serious Faults and Dangerous Faults
- Created a new function for working out Vehicle Checks Faults for Cat B 
- Created a new type FaultSummary and started to use it across the Cat B Journey

Tests haven't been updated as the functionality hasn't changed

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [ ] Code has been tested manually
- [ ] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

**Review by rossfauldsbjss — APPROVED** [review]
looks good

File: src/providers/fault-list/fault-list.ts [hunk 1]
```diff
@@ -48,100 +41,67 @@ export class FaultListProvider {
...
 
-  // Get Category B Faults
-
-  private getDrivingFaultsCatB (data: CatBUniqueTypes.TestData)
-  : (CommentedCompetency & MultiFaultAssignableCompetency)[] {
+  private getDrivingFaultsCatB (data: CatBUniqueTypes.TestData): FaultSummary[] {
     return [
-      ...this.getDrivingFaults(data.drivingFaults),
-      ...this.getManoeuvreFaults(data.manoeuvres, CompetencyOutcome.DF),
+      ...this.getComptencyFaults(data.drivingFaults),
```
Comment (ammarhaiderbjss) [inline src/providers/fault-list/fault-list.ts]: Typo spelling, "Competency"

File: src/shared/models/fault-marking.model.ts [hunk 2]
```diff
@@ -20,3 +20,5 @@ export enum CommentSource {
   EYESIGHT_TEST = 'eyesightTest',
   UNCOUPLE_RECOUPLE = 'uncoupleRecouple',
 }
+
+export type FaultSummary = Competency & MultiFaultAssignableCompetency & CommentedCompetency;
```
Comment (ammarhaiderbjss) [inline src/shared/models/fault-marking.model.ts:24]: nice!

**Comment by ammarhaiderbjss** [comment]
@Cavalier07 Have we refactored any logic here are are we just moving existing stuff around?

**Review by ammarhaiderbjss — APPROVED** [review]
Looks like a nice refactor! Might suggest a further improvement to add a `CompetencyIdentifiers` enum where we can hold all the strings that we do comparison checks on, e.g. "controlFault", "controlledStop" and "eyesightTest". Could also include suffixes as well for things like `endsWith(key, 'Fault')` and `key.endsWith('Comments'))`

**Comment by TomBillingtonUK** [comment]
> @Cavalier07 Have we refactored any logic here are we just moving existing stuff around?

I've moved the logic around and simplified some of it. There are 2 new methods (replacing 6 we used to have) but the strucutre and logic is very similar

**Comment by TomBillingtonUK** [comment]
> Looks like a nice refactor! Might suggest a further improvement to add a `CompetencyIdentifiers` enum where we can hold all the strings that we do comparison checks on, e.g. "controlFault", "controlledStop" and "eyesightTest". Could also include suffixes as well for things like `endsWith(key, 'Fault')` and `key.endsWith('Comments'))`

Done

File: src/shared/models/fault-marking.model.ts [hunk 3]
```diff
@@ -22,3 +22,11 @@ export enum CommentSource {
 }
 
 export type FaultSummary = Competency & MultiFaultAssignableCompetency & CommentedCompetency;
+
+export enum CompetencyIdentifiers {
+  COMMENTS = 'Comments',
```
Comment (ammarhaiderbjss) [inline src/shared/models/fault-marking.model.ts]: Sorry I know its pedantic but can we call this `COMMENTS_SUFFIX` and the fault one `FAULT_SUFFIX` just for clarity.

Comment (TomBillingtonUK) [reply src/shared/models/fault-marking.model.ts]: done

**Pull request merged**